sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -672,7 +672,11 @@ private[sql] object ParquetRelation2 {
         .getKeyValueMetaData
         .toMap
         .get(RowReadSupport.SPARK_METADATA_KEY)
-        .map(DataType.fromJson(_).asInstanceOf[StructType])
+        .map { serializedSchema =>
+          Try(DataType.fromJson(serializedSchema))
```

Comment:
Hm, after double thinking this, we may not need a log here. Because the user doesn't need to care about using JSON or old style case class string when reading/writing Parquet files. Spark SQL handles this automatically.